internal/clients/ec2/ec2_client.go

```diff
@@ -120,3 +120,34 @@ func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
 
 	return resp.InstanceTypes, nil
 }
+
+func (c *Client) RunInstances(ctx context.Context, amount int32, instanceType types.InstanceType, imageId string, keyName string) ([]*string, error) {
+	log.Trace().Msg("Run AWS EC2 instance")
+	input := &ec2.RunInstancesInput{
+		MaxCount:     aws.Int32(amount),
+		MinCount:     aws.Int32(amount),
```

Comment:
There isn't an `amount` field only min and max, and both are required. this is why it is written that way.

Reply:
it looks like it is a range, if the account can't run the max amount it will launch
the largest possible number of instances above MinCount

Reply:
I don't think that it is the functionality that we want for now. 
For now, we want a user to choose a specific amount of instances to provision. 
This is the way I found for now to enable that, but I guess that there is another one - I'll check it out.

Reply:
My guess is that this is for spot instances, for these types you define a price you are willing to pay and perhaps a range?

Reply:
After looking around didn't see any other options people used so maybe you are right and it is based on a budget range.